File: backend/src/db/projects/models/remains.py
from decimal import Decimal
from uuid import UUID

from sqlmodel import Field

from src.db.mixins import TimeStampWithIdMixin
from src.db.projects.models import ProjectsDataSQLModel


class RemainsBase(ProjectsDataSQLModel):
    cmo: str | None = Field(nullable=True)
    koc: int | None = Field(nullable=True)
    number: int | None = Field(nullable=True)
    indicator: int | None = Field(nullable=True)
    saldo_begin_debet: Decimal | None = Field(nullable=True)
    saldo_begin_credit: Decimal | None = Field(nullable=True)
    saldo_period_debet: Decimal | None = Field(nullable=True)
    saldo_period_credit: Decimal | None = Field(nullable=True)
    saldo_end_debet: Decimal | None = Field(nullable=True)
    saldo_end_credit: Decimal | None = Field(nullable=True)
    product_id: UUID | None = Field(nullable=True, foreign_key="products.id")  # TODO выкинуть нафиг
    company_id: UUID | None = Field(nullable=True, foreign_key="companies.id")


class Remains(RemainsBase, TimeStampWithIdMixin, table=True):
    __tablename__ = "remains"

File: frontend/src/Types/Remains/Remains.ts


export type ID = string;
export type Item = {
    "id": string;
    "created_at": string;
    "updated_at": string;
    "cmo": string;
    "koc": number,
    "number": number,
    "indicator": number,
    "saldo_begin_debet": number,
    "saldo_begin_credit": number,
    "saldo_period_debet": number,
    "saldo_period_credit": number,
    "saldo_end_debet": number,
    "saldo_end_credit": number,
    "product_id": string;
}